projects/packages/waf/src/class-waf-rules-manager.php

```diff
@@ -241,10 +269,10 @@ public static function generate_automatic_rules() {
 
 		try {
 			$rules = self::get_rules_from_api();
-		} catch ( \Exception $exception ) {
+		} catch ( Waf_Exception $e ) {
 			// Do not throw API exceptions for users who do not have access
```

Comment:
I know you did not come up with this in this PR, but how will users know then that they don't have access?

Reply:
I believe that the intention here was that if a site no longer has access to the latest firewall rules (read: no longer has a Jetpack Scan plan), we're still hitting the API and if we receive 401 response, we want to leave their rules file alone with whatever they already have. Which means that presently, the API dictates whether a site gets automatic rules, and we don't include a plan check on the client side, we just ask for them all the time.

Was added here: https://github.com/Automattic/jetpack/pull/24153

And relevant comment: https://github.com/Automattic/jetpack/pull/24153#issuecomment-1112539096

I think there is opportunity to improve the 401 handling here, and that could probably fall under "Minor error handling improvements" :+1: